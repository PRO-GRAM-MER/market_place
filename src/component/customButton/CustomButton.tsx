import React from "react";
import classes from "./customButton.module.css";

interface CustomButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
}

function CustomButton({
  children,
  className = "",
  ...props
}: CustomButtonProps) {
  return (
    <button
      role="button"
      className={`${classes.box} ${className}`.trim()} // Apply the CSS module's class
      {...props}
    >
      {children}
    </button>
  );
}

export default CustomButton;
