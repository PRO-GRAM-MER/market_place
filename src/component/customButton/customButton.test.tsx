import { render, fireEvent } from "@testing-library/react"; // Or from `vitest-browser-react` if needed for browser-specific testing
import CustomButton from "./CustomButton";
import classes from './customButton.module.css';
import { expect, test, vi } from "vitest";

test("renders the button with children", () => {
  const { getByRole } = render(<CustomButton>Click Me</CustomButton>);
  
  // Get the button by its role and accessible name
  const button = getByRole("button", { name: "Click Me" });

  // Assert that the button is in the document
  expect(button).toBeInTheDocument();
});

test("calls onClick when clicked", async () => {
  const onClick = vi.fn();
  const { getByRole } = render(<CustomButton onClick={onClick}>Click Me</CustomButton>);
  const button = getByRole("button", { name: "Click Me" });
  fireEvent.click(button);

  expect(onClick).toHaveBeenCalledTimes(1);
});

test("applies custom class name", () => {
  const customClass = "my-custom-class";
  const { getByRole } = render(<CustomButton className={customClass}>Click Me</CustomButton>);

  // Get the button by its role and name
  const button = getByRole("button", { name: "Click Me" });

  // Assert that the custom class is applied
  expect(button).toHaveClass(customClass);
});

test("applies default class name when no className is passed", () => {
  const { getByRole } = render(<CustomButton>Click Me</CustomButton>);

  const button = getByRole("button", { name: "Click Me" });
  expect(button.className).toContain(classes.box);
});


test("CustomButton snapshot test", () => {
  // Render the component
  const { asFragment } = render(<CustomButton>Click Me</CustomButton>);

  // Take a snapshot of the rendered component
  expect(asFragment()).toMatchSnapshot();
});