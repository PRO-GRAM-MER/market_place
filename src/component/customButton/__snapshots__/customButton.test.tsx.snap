// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomButton snapshot test 1`] = `
<DocumentFragment>
  <button
    class="undefined"
    role="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
